src/ducks/recurrence/utils.js

```diff
@@ -112,3 +115,53 @@ export const isDeprecatedBundle = recurrence => {
   const now = Date.now()
   return differenceInDays(now, latestDate)
 }
+
+export const addTransactionToOps = (bundles, transactions) => {
+  const updatedBundles = [...bundles].map(b => {
+    const bundle = { ...b }
+
+    // Matching on Amount, CategoryId and account
+    const transactionFounds = transactions.filter(transaction => {
+      return (
+        bundle.categoryIds.some(
+          catId => getCategoryId(transaction) === catId
+        ) &&
+        bundle.amounts.some(amount => amount === transaction.amount) &&
+        bundle.accounts?.some(account => account === transaction.account)
+      )
+    })
+
+    if (transactionFounds?.length > 0) {
+      bundle.ops = uniqBy([...bundle.ops, ...transactionFounds], o => o._id)
+    }
+
+    return bundle
+  })
+
+  return updatedBundles
+}
+
+const findBrandBundles = transactions => {
+  const brandBundles = transactions
+    .map(t => ({
+      ops: [t],
+      categoryIds: [getCategoryId(t)],
+      amounts: [t.amount]
+    }))
+    .map(brandSplit())
+    .map(b => b[0])
+    .filter(b => Boolean(b.brand))
+    .filter(b => Boolean(b.categoryIds[0]))
```

Comment:
So we exclude transactions without brand and at least one category? Having unit test for this one can be useful.

Reply:
old piece of code